Sources/HipMobileUI/ViewModels/Pages/AppetizerPageViewModel.cs

```diff
@@ -91,6 +100,46 @@ public AppetizerPageViewModel(Exhibit exhibit)
             await Navigation.PushAsync(downloadPage);
         }
 
+        /// <summary>
+        /// Sets the average user rating, the count and the stars if the user is connected to the internet.
+        /// </summary>
+        /// <returns></returns>
+        private async void SetUserRating()
+        {
+            if (IoCManager.Resolve<INetworkAccessChecker>().GetNetworkAccessStatus() == NetworkAccessStatus.NoAccess)
+            {
+                RatingAverage = "-";
+                RatingStars = "?????";
+                RatingCount = "- " + Strings.UserRating_Rate_Count;
+            }
+            else
+            {
+                UserRating userRating = await IoCManager.Resolve<IUserRatingManager>().GetUserRating(exhibit);
+                string stars = "";
+                for (int i = 1; i <= 5; i++)
+                {
+                    if (userRating.Average >= i)
+                        stars += "★";
+                    else
+                        stars += "☆";
+                }
+                if (userRating.Count > 0)
+                    RatingAverage = userRating.Average.ToString("0.#");
+                else
+                    RatingAverage = "-";
+                RatingStars = stars;
+                RatingCount = userRating.Count.ToString() + " " + Strings.UserRating_Rate_Count;
+            }
+        }
+
+        private async void GoToUserRatingPage()
+        {
+            if (IoCManager.Resolve<INetworkAccessChecker>().GetNetworkAccessStatus() != NetworkAccessStatus.NoAccess)
```

Comment:
And else? Maybe let the user know.

Reply:
Added a dialog for the user

Reply:
I don't see it here, only in another place?